vcd/resource_vcd_ip_space.go

```diff
@@ -180,9 +192,37 @@ func resourceVcdIpSpaceUpdate(ctx context.Context, d *schema.ResourceData, meta
 	ipSpaceConfig.ID = d.Id()
 	_, err = ipSpace.Update(ipSpaceConfig)
 	if err != nil {
+		// Duplicate key case can happen when splitting a single IP range
+		// start_address = "11.11.11.100"
+		// end_address   = "11.11.11.111"
+		// Into two ranges, where one matches `start_address` and another matches `end_address`
+		// start_address = "11.11.11.100" <- matches the start_address of single range I had
+		// end_address   = "11.11.11.108"
+		// start_address = "11.11.11.110"
+		// end_address   = "11.11.11.111" <- matches the end_address of single range I had
+		//
+		// It does not cause any issues on VCD or Terraform state except the split is not done even
+		// after multiple apply attempts.
+		// The point here is to hint user that what he is doing could be done in two steps
+		if strings.Contains(err.Error(), "Duplicate key") {
+			return diag.Errorf("error updating IP Space: splitting a single IP Range into two ranges "+
```

Comment:
This is really a helpful message. I wonder whether is it possible to tell the user which IP range is problematic?
Probably not as we only have `err` (and possibly it's enough with the `Error()` message), but just asking just in case 😄

Reply:
We probably can't be very precise, but VCD will hint. 

```
vcd_ip_space.space1: Modifying... [id=urn:vcloud:ipSpace:99746875-6942-4054-8030-77af980d9445]
╷
│ Error: error updating IP Space: splitting a single IP Range into two ranges with matching 'start_address' and 'end_address' is not supported in a single update operation. Please perform it in two steps: error updating IP Space: error in HTTP PUT request: INTERNAL_SERVER_ERROR - [ d20e9f9d-c985-4ef1-a60c-8b7be29d859b ] Duplicate key 7af38863-1af7-4481-9ee4-79c7ba782296 (attempted merging values 11.11.11.100 - 11.11.11.107 and 11.11.11.108 - 11.11.11.111)
│ 
│   with vcd_ip_space.space1,
│   on vcd.TestAccVcdIpSpaceIpAllocationstep2.tf line 45, in resource "vcd_ip_space" "space1":
│   45: resource "vcd_ip_space" "space1" {
```

A bit more in this discussion - https://github.com/vmware/terraform-provider-vcd/pull/1062#discussion_r1251898325

Reply:
That's great!